service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapperTest.java

```diff
@@ -95,7 +97,9 @@ public void tearDown() {
     @ParameterizedTest
     @MethodSource("resourceFileParams")
     public void When_MappingImmunizationJson_Expect_ObservationStatementXmlOutput(String inputJson, String outputXml,
-        boolean isNested) throws IOException {
+                                                                                  boolean isNested) throws IOException {
+        messageContext.getIdMapper().getOrNew(ResourceType.Practitioner, "6D340A1B-BC15-4D4E-93CF-BBCB5B74DF73");
```

Comment:
I don't think we have any tests covering the scenarios:
- immunization without a Practitioner
- immunization with a Practitioner, but without an actor